stubs/docutils/docutils/nodes.pyi

```diff
@@ -129,15 +137,64 @@ class Element(Node):
     def __add__(self, other: list[Node]) -> list[Node]: ...
     def __radd__(self, other: list[Node]) -> list[Node]: ...
     def __iadd__(self, other: Node | Iterable[Node]) -> Self: ...
-    def copy(self) -> Self: ...
-    def deepcopy(self) -> Self: ...
-    def pformat(self, indent: str = "    ", level: int = 0) -> str: ...
     def astext(self) -> str: ...
-    def index(self, item: Node, start: int = 0, stop: int = sys.maxsize) -> int: ...
-    def remove(self, item: Node) -> None: ...
+    def non_default_attributes(self) -> dict[str, Any]: ...
+    def attlist(self) -> list[tuple[str, Any]]: ...
+    def get(self, key: str, failobj: Any | None = None) -> Any: ...
+    def hasattr(self, attr: str) -> bool: ...
+    def delattr(self, attr: str) -> None: ...
+    def setdefault(self, key: str, failobj: Any | None = None) -> Any: ...
```

Comment:
hmm will need to think about this.

for `setdefault` if failobj is `T`, then the return type is `T`. if failobj is `None` then the return type is `T | None`, right?

so something like:

```python

_T = TypeVar("_T")

@overload
def setdefault(self, key: str, failobj: _T) -> _T: ...
@overload
def setdefault(self, key: str, failobj: None) -> _T | None: ...
def setdefault(self, key: str, failobj: _T | None) -> _T | None: ...
```

??

i don't think mypy will like `def setdefault(self, key: str, failobj: None) -> _T | None: ...` because the generic is only used once, so maybe it will have to be:

```python

_T = TypeVar("_T")

@overload
def setdefault(self, key: str, failobj: _T) -> _T: ...
@overload
def setdefault(self, key: str, failobj: None) -> Any | None: ...
def setdefault(self, key: str, failobj: _T | None) -> _T | None: ...
```

is that what you were thinking?

Reply:
and similar for 'get'?

```python
_T = TypeVar("_T")

@overload
def get(self, key: str, failobj: _T) -> _T: ...
@overload
def get(self, key: str, failobj: None) -> Any | None: ...
def get(self, key: str, failobj: _T | None) -> _T | None: ...
```